mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/metadata/DefaultComponentMetadataConfigurerFactory.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.internal.metadata;
+
+import org.mule.runtime.api.util.LazyValue;
+import org.mule.runtime.extension.api.metadata.ComponentMetadataConfigurer;
+import org.mule.runtime.extension.api.metadata.ComponentMetadataConfigurerFactory;
+
+import static java.util.ServiceLoader.load;
+
+public class DefaultComponentMetadataConfigurerFactory extends ComponentMetadataConfigurerFactory {
+
+  private static final LazyValue<ComponentMetadataConfigurerFactoryDelegate> DELEGATE =
+      new LazyValue(() -> load(ComponentMetadataConfigurerFactoryDelegate.class,
```

Comment:
add error handling. If none found, this won't return null but throw a NoSuchELementException.